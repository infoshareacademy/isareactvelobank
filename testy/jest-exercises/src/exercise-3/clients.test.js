// Napisz testy dla filterClients i getLatestRegisteredClient

import { filterClients, getLatestRegisteredClient } from './clients';
import { clients } from '../fixtures/clients-data';

describe('filterClients', () => {
    test('should filter INTERNAL clients', () => {
        const internalClients = filterClients(clients, { type: 'INTERNAL' });
        expect(internalClients).toHaveLength(3);
        expect(internalClients).toMatchSnapshot();
    });

    test('should filter EXTERNAL clients', () => {
        const externalClients = filterClients(clients, { type: 'EXTERNAL' });
        expect(externalClients).toHaveLength(2);
        expect(externalClients).toMatchSnapshot();
    });

    test('should not filter when filter type is unknown', () => {
        const filteredClients = filterClients(clients, {});
        expect(filteredClients).toBe(clients);
    });
});

describe('getLatestRegisteredClient', () => {
    test('should return client with latest registered date', () => {
        expect(getLatestRegisteredClient(clients).name).toEqual('Andrzej')
    });

    test('should return first client when there are two clients with the same date', () => {
        expect(getLatestRegisteredClient([
            { id: 5, name: 'Pierwszy Andrzej', type: 'EXTERNAL', registered: new Date(2020, 10, 9) },
            ...clients
        ]).name).toBe('Pierwszy Andrzej')
    });
});