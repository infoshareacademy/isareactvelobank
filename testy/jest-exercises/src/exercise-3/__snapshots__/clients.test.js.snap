// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filterClients should filter EXTERNAL clients 1`] = `
Array [
  Object {
    "id": 3,
    "name": "Anna",
    "registered": 2017-03-09T23:00:00.000Z,
    "type": "EXTERNAL",
  },
  Object {
    "id": 4,
    "name": "Andrzej",
    "registered": 2020-11-08T23:00:00.000Z,
    "type": "EXTERNAL",
  },
]
`;

exports[`filterClients should filter INTERNAL clients 1`] = `
Array [
  Object {
    "id": 1,
    "name": "Jan",
    "registered": 2019-05-11T22:00:00.000Z,
    "type": "INTERNAL",
  },
  Object {
    "id": 2,
    "name": "Maria",
    "registered": 2018-04-10T22:00:00.000Z,
    "type": "INTERNAL",
  },
  Object {
    "id": 5,
    "name": "Janina",
    "registered": 2019-12-02T23:00:00.000Z,
    "type": "INTERNAL",
  },
]
`;
